jwst/regtest/st_fitsdiff.py

```diff
@@ -1268,57 +1357,104 @@ def _diff(self):
 
-                # Add no more diff'd values than this
-                max_diffs = self.numdiffs - len(self.diff_values)
             else:
-                max_diffs = len(diffs[0])
-
-            last_seen_idx = None
-            for idx in islice(diffs[0], 0, max_diffs):
-                if idx == last_seen_idx:
-                    # Skip duplicate indices, which my occur when the column
-                    # data contains multi-dimensional values; we're only
-                    # interested in storing row-by-row differences
-                    continue
-                last_seen_idx = idx
-                self.diff_values.append(((col.name, idx), (arra[idx], arrb[idx])))
+                # The following lines include STScI's changes:
+                # - Calculate the absolute and relative differences separately for ad hoc report
+                # - If report_pixel_loc_diffs is False, just get the total differences,
+                #   it is not important where they come from
+
+                # Calculate the absolute and relative differences
+                nan_idx = np.isnan(arra) | np.isnan(arrb)
+                anonan = arra[~nan_idx]
+                bnonan = arrb[~nan_idx]
+                diffs = np.abs(anonan - bnonan)
+                abs_diffs = diffs[diffs > self.atol].size
+                nozeros = (diffs != 0.0) & (bnonan != 0.0)
+                rel_values = diffs[nozeros] / np.abs(bnonan[nozeros])
+                rel_diffs = rel_values[rel_values > self.rtol].size
+
+                if abs_diffs > 0 or rel_diffs > 0:
+                    # Report the total number of zeros, nans, and no-nan values
+                    self.report_zeros_nan.add_row(
+                        (
+                            col.name,
+                            arra[arra == 0.0].size,
+                            arrb[arrb == 0.0].size,
+                            arra[np.isnan(arra)].size,
+                            arrb[np.isnan(arrb)].size,
+                            arra[~np.isnan(arra)].size,
+                            arrb[~np.isnan(arrb)].size,
+                            np.mean(anonan),
+                            np.mean(bnonan),
+                        )
+                    )
+
+                    # Report the differences per column
+                    self.report_table.add_row(
+                        (
+                            col.name,
+                            abs_diffs,
+                            np.max(diffs),
+                            np.mean(diffs),
+                            np.std(diffs),
+                            rel_diffs,
+                            np.max(rel_values),
+                            np.mean(rel_values),
+                            np.std(rel_values),
+                        )
+                    )
 
+                self.diff_total += abs_diffs
+                self.rel_diffs += rel_diffs
```

Comment:
This needs to be indented to avoid reporting empty headers
```suggestion
                    self.diff_total += abs_diffs
                    self.rel_diffs += rel_diffs
```